openapi.yaml

```diff
@@ -300,6 +303,56 @@ paths:
       - events
 # {% if False %}
       - default
+# {% endif %}
+      x-codegen-request-body-name: body
+      x-openapi-router-controller: athenian.api.controllers.events_controller
+  /events/deployment/{name}/labels:
+    parameters:
+    - description: Name of the deployment.
+      in: path
+      name: name
+      required: true
+      schema:
+        type: string
+      style: simple
+    post:
+      operationId: insert_deployment_labels
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DeploymentInsertLabelsRequest'
+        x-body-name: body
+      responses:
+        200:
+          content:
+            application/json:
+              schema:
+                additionalProperties: false
+                properties:
+                  labels:
+                    $ref: '#/components/schemas/DeploymentLabels'
+                required:
+                - labels
+                type: object
+          description: The new labels associated with the deployment.
+        400:
+          $ref: '#/components/responses/InvalidRequestErrorResponse'
+        404:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GenericError'
+          description: A deployment with the given name was not found.
+      summary: |
+        Insert new labels for the deployment.
+        Existing labels with the same key will be replaced, other existing labels will not be modified.
```

Comment:
@gaetano-guerriero @vmarkovtsev why replacing labels here and not appending them? I'm thinking of a user that might want to map 2 issues successively to the same deployment, who would need to remember and store the 1st issue added when adding the 2nd label to not lose it.

Reply:
This would be the case with us where the system where we detect an issue associated with the deployment is potentially different from previously set labels so it would be hard to keep and share state between them. Not a big blocker for our use-case, though, as we could just keep the latest.

Reply:
@tjsousa after discussing with the team, we've decided to not invest the effort in supporting label appending for a given key. This would be quite expensive for us for a very specific use case. And since CFR comes down to counting deployments at the end of the day, this feature would not be strictly necessary to compute it. Hope that would still do the job for you.

If passing former labels for the same key is important, we can easily offer you a "read" endpoint to retrieve and pass the old ones too.